tempto-core/src/main/java/com/teradata/tempto/internal/fulfillment/table/jdbc/LoaderFactory.java

```diff
@@ -18,19 +18,30 @@
 import org.slf4j.Logger;
 
 import java.sql.JDBCType;
+import java.sql.ResultSet;
 import java.sql.SQLException;
+import java.util.ArrayList;
 import java.util.List;
+import java.util.Optional;
 
 import static org.slf4j.LoggerFactory.getLogger;
 
 class LoaderFactory
 {
     private static final Logger LOGGER = getLogger(LoaderFactory.class);
 
-    Loader create(QueryExecutor queryExecutor, String tableName)
+    Loader create(QueryExecutor queryExecutor, Optional<String> schema, String tableName)
             throws SQLException
     {
-        List<JDBCType> columnTypes = queryExecutor.executeQuery("SELECT * FROM " + tableName + " LIMIT 1").getColumnTypes();
+        if (!schema.isPresent() && queryExecutor.getConnector().isPresent() && queryExecutor.getConnector().get().equals("sqlserver")) {
```

Comment:
For SQL Server, I need to get the default schema, else getColumns won't work
getSchema was not working for Postgres - org.postgresql.jdbc4.Jdbc4Connection.getSchema() is not yet implemented